Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp

```diff
@@ -188,12 +211,97 @@ bool FileWatcher::Filter(QString path, const WatchRoot& watchRoot)
-        if (QFileInfo(path).isDir())
+    return true;
+}
+
+bool FileWatcher::IsExcluded(QString filepath) const
+{
+    for (const AssetBuilderSDK::FilePatternMatcher& matcher : m_excludes)
+    {
+        if (matcher.MatchesPath(filepath.toUtf8().constData()))
         {
-            return false;
+            return true;
         }
     }
-    return true;
+    return false;
+}
+
+void FileWatcher::AddExclusion(const AssetBuilderSDK::FilePatternMatcher& excludeMatch)
+{
+    m_excludes.push_back(excludeMatch);
+}
+
+void FileWatcher::InstallDefaultExclusionRules(QString cacheRootPath, QString projectRootPath)
+{
+    constexpr const char* intermediates = AssetProcessor::IntermediateAssetsFolderName;
+    using AssetBuilderSDK::FilePatternMatcher;
+    using AssetBuilderSDK::AssetBuilderPattern;
+
+    // Note to maintainers:
+    // If you add more here, consider updating DefaultExcludes_ExcludeExpectedLocations.  It turns out each platform
+    // can approach these exclusions slightly differently, due to slash direction, naming, and how it installs its
+    // file monitors.
+    // 
+    // File exclusions from the config are already checked on all files coming from the file watcher, but are done so
+    // in the main thread quite late in the process, so as not to block the file monitor unnecessarily.
+    // The file monitor is sensitive to being blocked due to it being a raw listener to some operating system level file event
+    // stream, and as little work in its threads should be done as possible, so do not add a large number of exclusions here.
+    // The best situation is a really small number of exclusions that match a very broad number of actual files (like the entire
+    // user folder full of log files).
+    //
+    // For most situations its probably better to just filter on the main thread instead of in the watcher - and most implementations
+    // of this class do just that.  
+    // However, on some operating systems, each monitored folder in a tree of monitored folders costs actual system resources
+    // (a handle) and there are limited handles available, so excluding entire folder trees that we know we don't care about
+    // is valuable to safe resources even if it costs more in the listener.
+    // 
+    // It is up to each implementation to make use of the list of excludes to best optimize itself for performance.  Even if the
+    // implementation does absolutely nothing with this exclude list, this class itself will still filter out excludes before
+    // forwarding the file events to actual handlers.
+    //
+    // To strike a balance here, add just a few hand-picked exclusions that tend to contain a lot of unnecessary folders:
+    //    * Everything in the cache EXCEPT for the "Intermediate Assets" and "fence" folders (filtering out fence will deadlock!)
+    //    * Project/build/*    (case insensitive)
```

Comment:
I actually started with that sort of thing ( I tried adding all the AP ignores) but the problem is that the watcher is very very sensitive to any slow down, we want only to absorb the absolute minimum of events here, and excluding everything at the watcher level isn't actually useful since the other parts of the program still re-run exclusions on anything already excluded here.   The exclusions here are mainly to keep the # of watched folders down.  I picked fence, user, and cache becuase htey provide thousands of excludes in just 1 exclude rule.